content/articles/adding-custom-methods-for-linq-queries/index.md

```diff
@@ -0,0 +1,161 @@
+### Adding custom methods for LINQ queries (C#) 
+### Introduction
+By introducing extension methods to the IEnumerableT> interface, you may expand the set of methods available for LINQ queries. Any data sequence can be converted into one single value by creating your custom aggregate method in SQL Server. 
+
+You can also build a method that returns a new series of values and acts as a custom filter or a specialized data transform for a sequence of variables. 
+
+A few examples include the Distinct and Skip techniques, as well as the Reverse.
+
+You may apply your methods to any enumerable collection when you extend the IEnumerableT> interface. In this article, we shall deeply take a look at various custom methods that are used in LINQ queries in C#.
+### Adding an aggregate method
+When a set of values is aggregated, a single value can be generated. Average, Min, and Max are just a few of the aggregate techniques available in LINQ. By introducing an extension method to the IEnumerableT> interface, you can design your aggregate method.
+
+Median is an extension method that computes the median of a sequence of double-valued numbers.
+```c#
+public static class LINQExtensionExample
+{
+    public static double Median(this IEnumerable<double>? source)
+    {
+        if (!(source?.Any() ?? false))
+        {
+            throw a new InvalidOperationException("A null or empty set cannot be used to compute the median..");
+        }
+
+        var sortedList = (from numberX in source
+                          orderby numberX
+                          select numberX).ToList();
+
+        int itemIndex = sortedList.Count / 2;
+
+        if (sortedList.Count % 2 == 0)
+        {
+            // Rerurning an even number of objects or items
+            return (sortedList[itemIndex] + sortedList[itemIndex - 1]) / 2;
+        }
+        else
+        {
+            // Rerurning an odd number of objects or items
+            return sortedList[itemIndex];
+        }
+    }
+}
+```
+There is no difference between calling this extension method from the IEnumerableT> interface and calling any other aggregate function.
+
+On a double array, the following code demonstrates using the Median method
+```c#
+double[] numbers1 = { 0.9, 5, 2, 9.3, 2.6, 7, 3.3, 4 };
+
+var query = numbers1.Median();
+
+Console.WriteLine("double: Median = " + query1);
+/*
+ The output of the above code is as shown below
+
+ Double: Median = 5.95
+*/
+```
+### An aggregate method is overloaded by accepting a variety of types.
+To accept sequences of different types, you can overload your aggregate method. Overloading each type is the typical technique. It's also possible to use a delegate to build an overload for a generic type. Combining both approaches is also an option.

```

Comment:
consider rewriting "It's also possible to use a delegate to build an overload for a generic type. Combining both approaches is also an option." to "However, it's also possible to use a delegate to build an overload for a generic type. Combining both approaches is also an option."